src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask4CreateConfigKeyField.java

```diff
@@ -0,0 +1,66 @@
+package com.semmle.jira.addon.config.upgrades;
+
+import com.atlassian.jira.component.ComponentAccessor;
+import com.atlassian.jira.issue.CustomFieldManager;
+import com.atlassian.jira.issue.context.GlobalIssueContext;
+import com.atlassian.jira.issue.fields.CustomField;
+import com.atlassian.plugin.spring.scanner.annotation.export.ExportAsService;
+import com.atlassian.sal.api.message.Message;
+import com.atlassian.sal.api.upgrade.PluginUpgradeTask;
+import com.google.common.collect.Iterables;
+import com.semmle.jira.addon.util.Constants;
+import com.semmle.jira.addon.util.JiraUtils;
+import java.util.Collection;
+import java.util.Collections;
+import org.springframework.stereotype.Component;
+
+@ExportAsService(PluginUpgradeTask.class)
+@Component
+public class UpgradeTask4CreateConfigKeyField implements PluginUpgradeTask {
+
+  private static final int BUILD_NUMER = 4;
+
+  @Override
+  public Collection<Message> doUpgrade() throws Exception {
+
+    CustomFieldManager customFieldManager = ComponentAccessor.getCustomFieldManager();
+
+    CustomField customField =
+        Iterables.getOnlyElement(
```

Comment:
getOnlyElement throws a NoSuchElementException if there is no element and IllegalArgumentException if there are more than one.

Reply:
Have you checked which plugin provides the `com.google.common.collect` package?

Reply:
Ah according to http://localhost:8080/plugins/servlet/upm/osgi the System Bundle exports that package, so it should be fairly safe to rely on it.